### PR TITLE
Added day of the week based automation.

### PR DESCRIPTION
**Pull request by Ekscentricitet**
Added day of the week based automation based on the existing code. Introduced a new struct Day and used it to transfer information from the listener to the pipeline trigger. Added subtitles for when this automation is chosen. The day of the week is localized, which might be a problem if, for example, the language of the OS is Bulgarian but the program is in English. Another approach would be to read the week from the resources but then we will have to add the 7 days of the week as a translation task for all languages.

### CODE REVIEW

**Comment by Ekscentricitet** [comment]
It seems like there are some changes that I did not make. I will investigate how they landed in the pull request and revert them.

**Comment by BartoszCichecki** [comment]
You most likely need to pull latest changes from my repo. I think GH allows you to sync with upstream repo.

**Comment by Ekscentricitet** [comment]
Alright, now there should be only my changes.

**Review by BartoszCichecki — CHANGES_REQUESTED** [review]
Good job, couple comments only: localization and cleaning up of the day/time situation. It's a little bit weird that Day also contains time and in many places both are used.

File: LenovoLegionToolkit.Lib.Automation/Pipeline/Triggers/IAutomationPipelineTrigger.cs [hunk 1]
```diff
@@ -49,6 +50,6 @@ public interface ITimeAutomationPipelineTrigger : IAutomationPipelineTrigger
     bool IsSunrise { get; }
     bool IsSunset { get; }
     Time? Time { get; }
-
-    ITimeAutomationPipelineTrigger DeepCopy(bool isSunrise, bool isSunset, Time? time);
+    Day? Day { get; }
```
[the remark this answers is not in the document]
Comment (Ekscentricitet) [reply LenovoLegionToolkit.Lib.Automation/Pipeline/Triggers/IAutomationPipelineTrigger.cs:53]: How do you propose to implement it? Should `Day` be removed as a struct and `DayOfWeek` be used instead? Then the `TimeAutomationEvent` would contain a `DateTime` object, while the `TimeAutomationPipelineTrigger` would have a `List<DayOfWeek> days` parameter. The Form will be passing both `Time` and `List<DayOfWeek>` when day automation is selected. And some checklist added to the UI.

File: LenovoLegionToolkit.WPF/Windows/Automation/TabItemContent/TimeAutomationPipelineTriggerTabItemContent.xaml [hunk 2]
```diff
@@ -59,5 +59,54 @@
...
+            GroupName="A" Checked="_dateRadioButton_Checked" />
+        <StackPanel
+            x:Name="_datePickerPanel"
+            Margin="28,0,0,0"
+            VerticalAlignment="Center"
+            IsEnabled="False"
+            Orientation="Horizontal">
+            <ComboBox 
+                x:Name="_datePickerDayOfWeek">
+                <ComboBoxItem Content="Monday" />
```
[the remark this answers is not in the document]
Comment (Ekscentricitet) [reply LenovoLegionToolkit.WPF/Windows/Automation/TabItemContent/TimeAutomationPipelineTriggerTabItemContent.xaml:76]: I localized the subtitles but didn't pay attention to these. Good catch!

File: LenovoLegionToolkit.Lib.Automation/Pipeline/Triggers/IAutomationPipelineTrigger.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (BartoszCichecki) [reply LenovoLegionToolkit.Lib.Automation/Pipeline/Triggers/IAutomationPipelineTrigger.cs:53]: I think that we can just store `DayOfWeek[]` just make sure that it's serialized and deserialized correctly. `TimeAutomationEvent` Can have two properties, you can just add `DayOfWeek[]` to the existing `Time` property.

```
public readonly struct TimeAutomationEvent : IAutomationEvent
{
    public Time Time { get; init; }
    public DayOfWeek[] Days { get; init; }
}
```

[the remark this answers is not in the document]
Comment (BartoszCichecki) [reply LenovoLegionToolkit.Lib.Automation/Pipeline/Triggers/IAutomationPipelineTrigger.cs:53]: Also, in `TimeAutomationListener` you can change the event to use a tuple:

`public event EventHandler<(Time, DayOfWeek[])>? Changed;`

This way we can easily pass the DayOfWeek together with time. I would assume that if the array is empty, event will apply to all days.

[the remark this answers is not in the document]
Comment (BartoszCichecki) [reply LenovoLegionToolkit.Lib.Automation/Pipeline/Triggers/IAutomationPipelineTrigger.cs:53]: I assume you know about tuple deconstruction? They are pretty cool to use. You can do

`var tuple = ("string", 1)`

and later you can deconstruct it:

`var (text, number) = tuple`.

Just a hint ;)

[the remark this answers is not in the document]
Comment (Ekscentricitet) [reply LenovoLegionToolkit.Lib.Automation/Pipeline/Triggers/IAutomationPipelineTrigger.cs]: I haven't worked with Touples much. I am not sure if they make a big difference here, but I used them for practice. Let me know if I did something incorrectly.

**Review by BartoszCichecki — APPROVED** [review]

**Comment by ZupoLlask** [comment]
Hi!

It seems this is what I was looking for: combine certain triggers with a specific day of week to change my battery charging mode.

What I don't get is how to use this?

I only find "at specific time", that doesn't work for me as I don't have a specific time to power on my PC.

Thanks! 🙏

**Pull request merged**